infra/modules/providers/azure/service-principal/main.tf

```diff
@@ -0,0 +1,25 @@
+module "azure-provider" {
+  source = "../provider"
+}
+
+locals {
+  rbac_true_false = "${var.create_for_rbac == "true" ? 1 : 0}"
+}
+
+data "azurerm_subscription" "sp" {}
+
+resource "azuread_application" "sp" {
+  count = "${local.rbac_true_false}"
+  name  = "${var.service_principal_display_name}"
+}
+
+resource "azuread_service_principal" "sp" {
+  count          = "${local.rbac_true_false}"
+  application_id = "${azuread_application.sp.application_id}"
+}
+
+resource "azurerm_role_assignment" "sp" {
```

Comment:
This supports the current "use case" where only a single `object_id` would need "rbac at the service principal level". Are there any use cases where would we need to support multiple `object_id`'s?

Reply:
We could certainly broaden that to loop(=TF12) or simulate a loop (<=TF11), but I'm happy with it this way at the moment.